addons/survey/static/tests/components/question_page_one2many_field.test.js

```diff
@@ -0,0 +1,288 @@
+import { expect, test } from "@odoo/hoot";
+import {
+    contains,
+    defineModels,
+    fields,
+    makeServerError,
+    models,
+    mountView,
+    onRpc,
+} from "@web/../tests/web_test_helpers";
+import { defineMailModels } from "@mail/../tests/mail_test_helpers";
+import { animationFrame, press, queryAll, queryOne } from "@odoo/hoot-dom";
+
+class Survey extends models.Model {
+    question_and_page_ids = fields.One2many({ relation: "survey_question" });
+    favorite_color = fields.Char({ string: "Favorite color" });
+
+    _records = [
+        {
+            id: 1,
+            question_and_page_ids: [1, 2],
+            favorite_color: "",
+        },
+    ];
+}
+
+class SurveyQuestion extends models.Model {
+    _name = "survey_question";
+
+    is_page = fields.Boolean();
+    title = fields.Char();
+    random_questions_count = fields.Integer({ string: "Question Count" });
+
+    _records = [
+        {
+            id: 1,
+            is_page: true,
+            title: "firstSectionTitle",
+            random_questions_count: 4,
+        },
+        {
+            id: 2,
+            is_page: false,
+            title: "recordTitle",
+            random_questions_count: 5,
+        },
+    ];
+    _views = {
+        "form,false": `
+            <form>
+                <field name="title" />
+            </form>
+        `,
+    };
+}
+
+defineModels([Survey, SurveyQuestion]);
+defineMailModels();
+
+test("basic rendering", async () => {
+    await mountView({
+        type: "form",
+        resModel: "survey",
+        resId: 1,
+        arch: `
+                <form>
+                    <field name="question_and_page_ids" widget="question_page_one2many">
+                        <list>
+                            <field name="is_page" invisible="1" />
+                            <field name="title" />
+                            <field name="random_questions_count" />
+                        </list>
+                    </field>
+                </form>
+            `,
+    });
+    expect(".o_field_x2many .o_list_renderer table.o_section_list_view").toHaveCount(1);
+    expect(".o_data_row").toHaveCount(2);
+    const rows = queryAll(".o_data_row");
+    expect(rows[0]).toHaveClass("o_is_section");
+    expect(rows[0]).toHaveClass("fw-bold");
```

Comment:
`toHaveClass` supports whitespace-separated strings or string arrays
```suggestion
    expect(rows[0]).toHaveClass("o_is_section fw-bold");
```